subcommand/health-sync/checks.go

```diff
@@ -161,8 +161,37 @@ func (c *Command) syncChecks(consulClient *api.Client,
 				currentStatuses[container.Name] = container.Health.Status
 			}
 		}
+		if container.Name == config.ConsulDataplaneContainerName {
+			if container.Health.Status == ecs.HealthStatusHealthy {
+				overallDataplaneHealth = 1
+			}
+		}
 	}
 
+	for containerName, healthStatus := range parsedContainers {
+
+		if containerName != config.ConsulDataplaneContainerName {
+			currentContainerHealth := 0
+			if healthStatus == ecs.HealthStatusHealthy {
+				currentContainerHealth = 1
+			}
+			overallDataplaneHealth = overallDataplaneHealth & currentContainerHealth
+		}
+	}
+	overallDataplaneHealthStatus := ecs.HealthStatusUnhealthy
+	if overallDataplaneHealth == 1 {
+		overallDataplaneHealthStatus = ecs.HealthStatusHealthy
+	}
```

Comment:
I think the logic here can be simplified to the following:
```suggestion
        overallDataplaneHealthStatus := ecs.HealthStatusHealthy
	for containerName, healthStatus := range parsedContainers {
			if healthStatus != ecs.HealthStatusHealthy {
				overallDataplaneHealthStatus = ecs.HealthStatusUnhealthy
				break
			}
		}
	}
```

Reply:
I tried this but we need to use dataplane container health as well to decide the overall status. This results in failure of the case when daplane container itself is missing.

Reply:
If my assumptions are right the code would look something along those lines:
```suggestion
	overallDataplaneHealthStatus, ok := parsedContainers[config.ConsulDataplaneContainerName]
	// if dataplane container exist and healthy then proceed to checking the other containers health
	if ok && overallDataplaneHealthStatus == ecs.HealthStatusHealthy {
		//
		for name, healthStatus := range parsedContainers {
			// as soon as we find a healthy container, we can set the health the healthy
			if healthStatus == ecs.HealthStatusHealthy && name != config.ConsulDataplaneContainerName {
				overallDataplaneHealthStatus = ecs.HealthStatusUnhealthy
				break
			}
		}
	} else {
		// If no dataplane container or dataplane container is not healthy set overall health to healthy
		overallDataplaneHealthStatus = ecs.HealthStatusUnhealthy
	}
```

Reply:
Actually, we need to check all the other containers.
It's not atleast one which needs to be healthy but if any one of the containers is unhealthy, we need to mark the dataplane as unhealthy. 
 
 Effectively what we are saying is:
 1. If dataplane is unhealthy - UNHEALTHY
 2. if dataplane is healthy - 
 - if all service containers are healthy - HEALTHY
 - if any one of the service containers are unhealthy - UNHEALTHY 
 
 Let me know if this makes sense. I had referred the tests to understand these scenarios